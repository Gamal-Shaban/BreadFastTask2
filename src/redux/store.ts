import {configureStore} from '@reduxjs/toolkit';
import appStateReducer from './slices/appStateSlice';
import postsReducer from './slices/postsSlice';
import commentsReducer from './slices/commetsSlice';

export const store = configureStore({
  reducer: {
    appState: appStateReducer,
    posts: postsReducer,
    comments: commentsReducer,
  },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
