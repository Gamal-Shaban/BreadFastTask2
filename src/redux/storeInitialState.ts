import {initialStatesType} from '../constants';

const initialStates: initialStatesType = {
  appState: {
    isLoaded: false,
  },
};

export default initialStates;
