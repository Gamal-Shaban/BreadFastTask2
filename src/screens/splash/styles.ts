import {StyleSheet} from 'react-native';
import {COLORS} from '../../theme/theme';
import {horizontalScale, verticalScale} from '../../utils/responsiveUtils';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS.WHITE,
  },
  image: {
    width: horizontalScale(125),
    height: verticalScale(125),
    marginTop: verticalScale(-35),
    alignSelf: 'center',
  },
});
