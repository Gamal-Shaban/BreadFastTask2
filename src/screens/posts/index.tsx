import React, {useEffect} from 'react';
import {FlatList, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {AppDispatch} from '../../redux/store';
import {initialStatesType, PostDetailsType} from '../../types';
import {PostItem} from '../../components/postItem';
import {fetchPosts, setSelectedPost} from '../../redux/slices/postsSlice';
import {Loading} from '../../components/loading';
import {styles} from './styles';
import {useNavigation} from '@react-navigation/native';

export const PostsScreen = () => {
  const dispatch = useDispatch<AppDispatch>();
  const {postsData, loading} = useSelector((state: initialStatesType) => ({
    postsData: state?.posts?.postsData,
    loading: state?.posts?.isLoading,
  }));
  const {navigate} = useNavigation();

  useEffect(() => {
    dispatch(fetchPosts());
  }, [dispatch]);

  const renderItem = ({item}: {item: PostDetailsType}) => {
    return <PostItem item={item} onPressItem={onPressItem} />;
  };
  const onPressItem = (item: PostDetailsType) => {
    dispatch(setSelectedPost(item));
    navigate(
      'postsDetails' as never,
      {
        postId: item.id,
      } as never,
    );
  };

  return (
    <View style={styles.container}>
      {loading ? (
        <Loading />
      ) : (
        <FlatList
          data={postsData}
          renderItem={renderItem}
          contentContainerStyle={styles.contentContainerStyle}
        />
      )}
    </View>
  );
};
