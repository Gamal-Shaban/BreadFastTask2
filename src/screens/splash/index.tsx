import React from 'react';
import {View} from 'react-native';
import {styles} from './styles';

export const SplashScreen = () => {
  return <View style={styles.container} />;
};
