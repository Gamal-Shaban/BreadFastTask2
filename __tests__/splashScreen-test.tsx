/**
 * @format
 */

import 'react-native';
import React from 'react';
import {store} from '../src/redux/store';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {SplashScreen} from '../src/screens/splash';

// Note: test renderer must be required after react-native.

it('should Renders Splash and MatchSnapShot', () => {
  const {toJSON} = render(
    <Provider store={store}>
      <SplashScreen />
    </Provider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
