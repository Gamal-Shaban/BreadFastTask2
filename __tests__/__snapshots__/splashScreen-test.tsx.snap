// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Renders Splash and MatchSnapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/loading.gif",
      }
    }
  />
</View>
`;
